### PR TITLE
refactor stf task receiver

### LINKED ISSUE #106: refactor stf-task receiver client to more generic style
as the title.

### PR DESCRIPTION
**Pull request by BillyWooo**
resolves #106

### CODE REVIEW

File: app-libs/stf/src/stf_sgx_litentry.rs [hunk 1]
```diff
@@ -130,6 +130,16 @@ impl Stf {
 		}
 	}
 
+	pub fn verify_ruleset2(who: AccountId) -> StfResult<()> {
```
Comment (Kailai-Wang) [inline app-libs/stf/src/stf_sgx_litentry.rs:133]: I thought ruleset 2 is about discord verification and server joining?

**Review by Kailai-Wang — COMMENTED** [review]
Can we delay this PR a bit? Based on the following reasons:

- I'm currently working on web3-identity-verification which has a different structure
- the web2/3 verification and ruleset-building will be moved out of receiver, because not all tasks need to go through the sender/receiver channel (before we add more generic callback trait), I think separate crates under `core/` would be better
- web3 verification doesn't need a httpclient
- maybe `MakeClient` is not enough for ruleset building, imagine we have to build a ruleset where multiple http requests to different servers are needed.

My imaginary structure about `stf-task` + `identity-verification` + `ruleset-build`: they are all in parallel:

image

File: app-libs/stf/src/stf_sgx_litentry.rs (hunk 1 above)
Comment (BillyWooo) [reply app-libs/stf/src/stf_sgx_litentry.rs:133]: Oh, sorry. This file should be exclude. Forgot.

**Comment by BillyWooo** [comment]
> Can we delay this PR a bit? Based on the following reasons:
> 
> * I'm currently working on web3-identity-verification which has a different structure
> * the web2/3 verification and ruleset-building will be moved out of receiver, because not all tasks need to go through the sender/receiver channel (before we add more generic callback trait), I think separate crates under `core/` would be better
> * web3 verification doesn't need a httpclient
> * maybe `MakeClient` is not enough for ruleset building, imagine we have to build a ruleset where multiple http requests to different servers are needed.
> 
> My imaginary structure about `stf-task` + `identity-verification` + `ruleset-build`: they are all in parallel:
> 
> image

Ok. Let's wait a bit.
The discussion with Zhiming, we thought all verifications can go through http. So we move it this way. But never mind, let's wait.

**Comment by BillyWooo** [comment]
obsolete, because of #89

**Pull request closed without merging**